conductr_cli/bndl_main.py

```diff
@@ -35,6 +38,40 @@ def run(argv=None):
         sys.exit(args.func(args))
 
 
+class EndpointAction(argparse.Action):
+    def __call__(self, parser, namespace, value, option_strings):
+        log = logging.getLogger(__name__)
+        if option_strings == '--endpoint':
+            namespace.endpoint_dicts.append({'name': value})
+        elif not namespace.endpoint_dicts:
+            log.error('bndl: argument {} needs to be specified after the argument --endpoint'
+                      .format(option_strings))
+            sys.exit(2)
+        elif option_strings == '--acl':
+            last_endpoint = namespace.endpoint_dicts[-1]
+            if 'acls' in last_endpoint:
+                last_endpoint['acls'].append(value)
+            else:
+                last_endpoint['acls'] = [value]
+        else:
+            dict_key = option_strings[2:]
+            namespace.endpoint_dicts[-1][dict_key] = value
+
+
+def set_endpoints(args):
+    def validate(endpoint):
+        if 'component' not in endpoint:
+            log.error('bndl: argument --component is required when specifying argument --endpoint')
```

Comment:
Perhaps also display the name of the endpoint specified by `--endpoint` - this will help the user narrow down which endpoint argument is causing this error when there are multiple `--endpoint` specified.